.github/workflows-helpers/run-e2e-test.sh

```diff
@@ -0,0 +1,55 @@
+# Check if the pod with the matching image SHA and purpose is ready
```

Comment:
Not a blocker but just a question.

Is the plan to keep this bash script as is? I know you've used things like argoCD and other DAG like workflow builders, so was just surprised to have a simple bash script like it today.